### PR TITLE
[gpx] Remove duplicated code

### PR DESCRIPTION
**Pull request by cyber-toad**

### CODE REVIEW

**Review by vng — COMMENTED** [review]
Well, good, but I'd prefer to make serdes_common.cpp module for that.

File: kml/serdes_common.hpp [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
 auto constexpr kDefaultTrackWidth = 5.0;
 auto constexpr kDefaultTrackColor = 0x006ec7ff;
 
-template <typename Channel>
-uint32_t ToRGBA(Channel red, Channel green, Channel blue, Channel alpha)
-{
-  return static_cast<uint8_t>(red) << 24 | static_cast<uint8_t>(green) << 16 |
-         static_cast<uint8_t>(blue) << 8 | static_cast<uint8_t>(alpha);
-}
+template <typename Channel> uint32_t ToRGBA(Channel red, Channel green, Channel blue, Channel alpha);
```
Comment (vng) [inline kml/serdes_common.hpp:15]: This template function remains in hpp

File: kml/serdes_common.hpp [hunk 2]
```diff
@@ -1,18 +1,21 @@
 #pragma once
 
+#include "base/string_utils.hpp"
```
Comment (vng) [inline kml/serdes_common.hpp:3]: Move to cpp

File: kml/serdes_common.hpp [hunk 3]
```diff
@@ -1,18 +1,21 @@
...
 #include "coding/string_utf8_multilang.hpp"
+#include "geometry/point2d.hpp"
+#include "geometry/point_with_altitude.hpp"
+#include <sstream>
```
Comment (vng) [inline kml/serdes_common.hpp:7]: Move sstream to cpp

File: kml/serdes_common.cpp [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+std::string PointToString(geometry::PointWithAltitude const & pt)
+{
+  if (pt.GetAltitude() != geometry::kInvalidAltitude)
+    return PointToString(pt.GetPoint()) + "," + strings::to_string(pt.GetAltitude());
+  return PointToString(pt.GetPoint());
+}
+
+
+}
```
Comment (vng) [inline kml/serdes_common.cpp:34]: // namespace kml

**Review by vng — COMMENTED** [review]
Add serdes_common.cpp to xcode/kml/kml.xcodeproj

File: kml/serdes_common.hpp (hunk 3 above)
Comment (cyber-toad) [reply kml/serdes_common.hpp:7]: Fixed

File: kml/serdes_common.hpp (hunk 2 above)
Comment (cyber-toad) [reply kml/serdes_common.hpp:3]: Fixed

File: kml/serdes_common.hpp (hunk 1 above)
Comment (cyber-toad) [reply kml/serdes_common.hpp:15]: Fixed

**Comment by cyber-toad** [comment]
> Add serdes_common.cpp to xcode/kml/kml.xcodeproj

@vng is it possible to do on Linux? Is this project file generated somehow by cmake or xcode is mandatory?

**Comment by vng** [comment]
PTAL

**Review by vng — APPROVED** [review]

File: kml/serdes_common.cpp [hunk 5]
```diff
@@ -0,0 +1,28 @@
+#include "base/string_utils.hpp"
+#include "kml/serdes_common.hpp"
+#include "geometry/mercator.hpp"
+#include <sstream>
+
+namespace kml
+{
+
+std::string PointToString(m2::PointD const & org)
```
Comment (biodranik) [inline kml/serdes_common.cpp:9]: Are points with elevation logged properly? If not, then is todo needed?

**Review by biodranik — APPROVED** [review]

**Pull request merged**